# Car Market Analysis on eBay Kleinanzeigen

The aim of this project is to explore dataset of used cars from *eBay Kleinanzeigen*, a classifieds section of the German eBay website.

The data dictionary provided with data is as follows:

* `dateCrawled` - When this ad was first crawled.
* `All field` -values are taken from this date.
* `name` - Name of the car.
* `seller` - Whether the seller is private or a dealer.
* `offerType` - The type of listing. 
* `price` - The price on the ad to sell the car.
* `abtest` - Whether the listing is included in an A/B test.
* `vehicleType` - The vehicle Type.
* `yearOfRegistration` - The year in which the car was first registered.
* `gearbox` - The transmission type.
* `powerPS` - The power of the car in PS.
* `model` - The car model name.
* `odometer` - How many kilometers the car has driven.
* `monthOfRegistration` - The month in which the car was first registered.
* `fuelType` - What type of fuel the car uses.
* `brand` - The brand of the car.
* `notRepairedDamage` - If the car has a damage which is not yet repaired.
* `dateCreated` - The date on which the eBay listing was created.
* `nrOfPictures` - The number of pictures in the ad.
* `postalCode` - The postal code for the location of the vehicle.
* `lastSeenOnline` - When the crawler saw this ad last online.

In [44]:
import pandas as pd
import numpy as np

autos = pd.read_csv("autos.csv", encoding= "Latin-1")
autos.info()
autos.head()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 371528 entries, 0 to 371527
Data columns (total 20 columns):
 #   Column               Non-Null Count   Dtype 
---  ------               --------------   ----- 
 0   dateCrawled          371528 non-null  object
 1   name                 371528 non-null  object
 2   seller               371528 non-null  object
 3   offerType            371528 non-null  object
 4   price                371528 non-null  int64 
 5   abtest               371528 non-null  object
 6   vehicleType          333659 non-null  object
 7   yearOfRegistration   371528 non-null  int64 
 8   gearbox              351319 non-null  object
 9   powerPS              371528 non-null  int64 
 10  model                351044 non-null  object
 11  kilometer            371528 non-null  int64 
 12  monthOfRegistration  371528 non-null  int64 
 13  fuelType             338142 non-null  object
 14  brand                371528 non-null  object
 15  notRepairedDamage    299468 non-nu

dateCrawled                            name  seller offerType  \
0  2016-03-24 11:52:17                      Golf_3_1.6  privat   Angebot   
1  2016-03-24 10:58:45            A5_Sportback_2.7_Tdi  privat   Angebot   
2  2016-03-14 12:52:21  Jeep_Grand_Cherokee_"Overland"  privat   Angebot   
3  2016-03-17 16:54:04              GOLF_4_1_4__3TÜRER  privat   Angebot   
4  2016-03-31 17:25:20  Skoda_Fabia_1.4_TDI_PD_Classic  privat   Angebot   

   price abtest vehicleType  yearOfRegistration    gearbox  powerPS  model  \
0    480   test         NaN                1993    manuell        0   golf   
1  18300   test       coupe                2011    manuell      190    NaN   
2   9800   test         suv                2004  automatik      163  grand   
3   1500   test  kleinwagen                2001    manuell       75   golf   
4   3600   test  kleinwagen                2008    manuell       69  fabia   

   kilometer  monthOfRegistration fuelType       brand notRepairedDamage  \
0     150000                    0   benzin  volkswagen               NaN   
1     125000                    5   diesel        audi                ja   
2     125000                    8   diesel        jeep               NaN   
3     150000                    6   benzin  volkswagen              nein   
4      90000                    7   diesel       skoda              nein   

           dateCreated  nrOfPictures  postalCode             lastSeen  
0  2016-03-24 00:00:00             0       70435  2016-04-07 03:16:57  
1  2016-03-24 00:00:00             0       66954  2016-04-07 01:46:50  
2  2016-03-14 00:00:00             0       90480  2016-04-05 12:47:46  
3  2016-03-17 00:00:00             0       91074  2016-03-17 17:40:17  
4  2016-03-31 00:00:00             0       60437  2016-04-06 10:17:21

## Initial Observations
Based on the work above, we have the following observations:

* The dataset contains 20 columns, most of which are objects.
* Some columns have null values, but none have more than 20% of null values
* The columns' names use camelcase instead of snakecase so we can't just remove the underscore.

## Change columns' names

We markdown a few changes here:
1. Change the columns from camelcase to snakecase
2. Change a few wordings to describe the columns more accurrately


In [45]:
autos.columns


Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'kilometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

In [46]:
autos.columns = ['date_crawled', 'name', 'seller', 'offer_type', 'price', 'abtest',
       'vehicle_type', 'registration_year', 'gear_box', 'power_ps', 'model',
       'km', 'registration_month', 'fuel_type', 'brand',
       'unrepaired_damage', 'date_ad', 'number_of_pictures', 'postal_code',
       'last_seen']
autos.columns

Index(['date_crawled', 'name', 'seller', 'offer_type', 'price', 'abtest',
       'vehicle_type', 'registration_year', 'gear_box', 'power_ps', 'model',
       'km', 'registration_month', 'fuel_type', 'brand', 'unrepaired_damage',
       'date_ad', 'number_of_pictures', 'postal_code', 'last_seen'],
      dtype='object')

## Cleaning Data

There are some observations we have and what other cleaning tasks needed to be done:

In [47]:
autos.describe(include='all') # We use "all" to include both categoric and numeric numbers


date_crawled         name  seller offer_type         price  \
count                371528       371528  371528     371528  3.715280e+05   
unique               280500       233531       2          2           NaN   
top     2016-03-24 14:49:47  Ford_Fiesta  privat    Angebot           NaN   
freq                      7          657  371525     371516           NaN   
mean                    NaN          NaN     NaN        NaN  1.729514e+04   
std                     NaN          NaN     NaN        NaN  3.587954e+06   
min                     NaN          NaN     NaN        NaN  0.000000e+00   
25%                     NaN          NaN     NaN        NaN  1.150000e+03   
50%                     NaN          NaN     NaN        NaN  2.950000e+03   
75%                     NaN          NaN     NaN        NaN  7.200000e+03   
max                     NaN          NaN     NaN        NaN  2.147484e+09   

        abtest vehicle_type  registration_year gear_box       power_ps  \
count   371528       333659      371528.000000   351319  371528.000000   
unique       2            8                NaN        2            NaN   
top       test    limousine                NaN  manuell            NaN   
freq    192585        95894                NaN   274214            NaN   
mean       NaN          NaN        2004.577997      NaN     115.549477   
std        NaN          NaN          92.866598      NaN     192.139578   
min        NaN          NaN        1000.000000      NaN       0.000000   
25%        NaN          NaN        1999.000000      NaN      70.000000   
50%        NaN          NaN        2003.000000      NaN     105.000000   
75%        NaN          NaN        2008.000000      NaN     150.000000   
max        NaN          NaN        9999.000000      NaN   20000.000000   

         model             km  registration_month fuel_type       brand  \
count   351044  371528.000000       371528.000000    338142      371528   
unique     251            NaN                 NaN         7          40   
top       golf            NaN                 NaN    benzin  volkswagen   
freq     30070            NaN                 NaN    223857       79640   
mean       NaN  125618.688228            5.734445       NaN         NaN   
std        NaN   40112.337051            3.712412       NaN         NaN   
min        NaN    5000.000000            0.000000       NaN         NaN   
25%        NaN  125000.000000            3.000000       NaN         NaN   
50%        NaN  150000.000000            6.000000       NaN         NaN   
75%        NaN  150000.000000            9.000000       NaN         NaN   
max        NaN  150000.000000           12.000000       NaN         NaN   

       unrepaired_damage              date_ad  number_of_pictures  \
count             299468               371528            371528.0   
unique                 2                  114                 NaN   
top                 nein  2016-04-03 00:00:00                 NaN   
freq              263182                14450                 NaN   
mean                 NaN                  NaN                 0.0   
std                  NaN                  NaN                 0.0   
min                  NaN                  NaN                 0.0   
25%                  NaN                  NaN                 0.0   
50%                  NaN                  NaN                 0.0   
75%                  NaN                  NaN                 0.0   
max                  NaN                  NaN                 0.0   

         postal_code            last_seen  
count   371528.00000               371528  
unique           NaN               182806  
top              NaN  2016-04-07 06:45:59  
freq             NaN                   17  
mean     50820.66764                  NaN  
std      25799.08247                  NaN  
min       1067.00000                  NaN  
25%      30459.00000                  NaN  
50%      49610.00000                  NaN  
75%      71546.00000           

Our initial observations:
There are a number of columns that have the same values (seller, offer_type columns) that we can drop to make dataset more clean. And, the number_of_pictures look a little odd so we need to investigate further:

It looks like the number_of_pictures has 0 for every row so we need to drop this 

In [48]:
autos = autos.drop(["number_of_pictures", "seller", "offer_type"], axis=1)
autos.info()



<class 'pandas.core.frame.DataFrame'>
RangeIndex: 371528 entries, 0 to 371527
Data columns (total 17 columns):
 #   Column              Non-Null Count   Dtype 
---  ------              --------------   ----- 
 0   date_crawled        371528 non-null  object
 1   name                371528 non-null  object
 2   price               371528 non-null  int64 
 3   abtest              371528 non-null  object
 4   vehicle_type        333659 non-null  object
 5   registration_year   371528 non-null  int64 
 6   gear_box            351319 non-null  object
 7   power_ps            371528 non-null  int64 
 8   model               351044 non-null  object
 9   km                  371528 non-null  int64 
 10  registration_month  371528 non-null  int64 
 11  fuel_type           338142 non-null  object
 12  brand               371528 non-null  object
 13  unrepaired_damage   299468 non-null  object
 14  date_ad             371528 non-null  object
 15  postal_code         371528 non-null  int64 
 16  la

There are two columns, price and auto, which are numeric values with extra characters being stored as text. We'll clean and convert these.


In [49]:
autos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 371528 entries, 0 to 371527
Data columns (total 17 columns):
 #   Column              Non-Null Count   Dtype 
---  ------              --------------   ----- 
 0   date_crawled        371528 non-null  object
 1   name                371528 non-null  object
 2   price               371528 non-null  int64 
 3   abtest              371528 non-null  object
 4   vehicle_type        333659 non-null  object
 5   registration_year   371528 non-null  int64 
 6   gear_box            351319 non-null  object
 7   power_ps            371528 non-null  int64 
 8   model               351044 non-null  object
 9   km                  371528 non-null  int64 
 10  registration_month  371528 non-null  int64 
 11  fuel_type           338142 non-null  object
 12  brand               371528 non-null  object
 13  unrepaired_damage   299468 non-null  object
 14  date_ad             371528 non-null  object
 15  postal_code         371528 non-null  int64 
 16  la

In [50]:
print(autos["km"].describe())
print(autos["km"].shape)

count    371528.000000
mean     125618.688228
std       40112.337051
min        5000.000000
25%      125000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: km, dtype: float64
(371528,)


In [51]:
autos["km"].value_counts().sort_index(ascending=False).head(15)

150000    240797
125000     38067
100000     15920
90000      12523
80000      11053
70000       9773
60000       8669
50000       7615
40000       6376
30000       6041
20000       5676
10000       1949
5000        7069
Name: km, dtype: int64

We can tell then the dispersion of values looks realistic. Now let's check `price` column.

In [52]:
print(autos["price"].unique().shape)
print(autos["price"].describe())
autos["price"].value_counts().head(10)
                                


(5597,)
count    3.715280e+05
mean     1.729514e+04
std      3.587954e+06
min      0.000000e+00
25%      1.150000e+03
50%      2.950000e+03
75%      7.200000e+03
max      2.147484e+09
Name: price, dtype: float64


0       10778
500      5670
1500     5394
1000     4649
1200     4594
2500     4438
600      3819
3500     3792
800      3784
2000     3432
Name: price, dtype: int64

There are 5597 uniqe values in 'price' column. As there are a lot of uniqe values, we can not check them all at once and actually we do not need to. Our field of intrest are outliers, so we can sort the price in ascending and descending order and then inspect first 30 results at the top and at the bottom of the list.

In [53]:
autos["price"].value_counts().sort_index(ascending=False).head(30) #Check the number of each value in ascending order

2147483647     1
99999999      15
99000000       1
74185296       1
32545461       1
27322222       1
14000500       1
12345678       9
11111111      10
10010011       1
10000000       8
9999999        3
3895000        1
3890000        1
2995000        1
2795000        1
1600000        2
1300000        1
1250000        2
1234566        1
1111111        2
1010010        1
1000000        5
999999        13
999990         1
911911         1
849000         1
820000         1
780000         1
745000         2
Name: price, dtype: int64

In [54]:
autos["price"].value_counts().sort_index(ascending=False).tail(30) 

35       18
33        1
32        1
30       55
29        2
27        1
26        1
25       33
24        1
21        1
20       51
19        3
18        3
17        5
16        2
15       27
14        5
13        7
12        8
11        5
10       84
9         8
8         9
7         3
5        26
4         1
3         8
2        12
1      1189
0     10778
Name: price, dtype: int64

We've checked 30 first unique elements in p`price` column sorted in descending order and can tell that it is unlikely that car price could be less then 100-200 eur. Analysing of the top 30 of the most expencive cars shows us the highest prices more then 2 mln USD, or like 999990 for used car seem odd too. We can assume the criteria that all the prices more the 350 000 USD (the next price after 999990 in our list) and less then 200 USD can be removed from the dataset.

In [55]:
autos[autos["price"].between(200, 350000)]

date_crawled                                          name  \
0       2016-03-24 11:52:17                                    Golf_3_1.6   
1       2016-03-24 10:58:45                          A5_Sportback_2.7_Tdi   
2       2016-03-14 12:52:21                Jeep_Grand_Cherokee_"Overland"   
3       2016-03-17 16:54:04                            GOLF_4_1_4__3TÜRER   
4       2016-03-31 17:25:20                Skoda_Fabia_1.4_TDI_PD_Classic   
...                     ...                                           ...   
371523  2016-03-14 17:48:27                    Suche_t4___vito_ab_6_sitze   
371524  2016-03-05 19:56:21         Smart_smart_leistungssteigerung_100ps   
371525  2016-03-19 18:57:12            Volkswagen_Multivan_T4_TDI_7DC_UY2   
371526  2016-03-20 19:41:08                        VW_Golf_Kombi_1_9l_TDI   
371527  2016-03-07 19:39:19  BMW_M135i_vollausgestattet_NP_52.720____Euro   

        price   abtest vehicle_type  registration_year   gear_box  power_ps  \
0         480     test          NaN               1993    manuell         0   
1       18300     test        coupe               2011    manuell       190   
2        9800     test          suv               2004  automatik       163   
3        1500     test   kleinwagen               2001    manuell        75   
4        3600     test   kleinwagen               2008    manuell        69   
...       ...      ...          ...                ...        ...       ...   
371523   2200     test          NaN               2005        NaN         0   
371524   1199     test       cabrio               2000  automatik       101   
371525   9200     test          bus               1996    manuell       102   
371526   3400     test        kombi               2002    manuell       100   
371527  28990  control    limousine               2013    manuell       320   

              model      km  registration_month fuel_type           brand  \
0              golf  150000                   0    benzin      volkswagen   
1               NaN  125000                   5    diesel            audi   
2             grand  125000                   8    diesel            jeep   
3              golf  150000                   6    benzin      volkswagen   
4             fabia   90000                   7    diesel           skoda   
...             ...     ...                 ...       ...             ...   
371523          NaN   20000                   1       NaN  sonstige_autos   
371524       fortwo  125000                   3    benzin           smart   
371525  transporter  150000                   3    diesel      volkswagen   
371526         golf  150000                   6    diesel      volkswagen   
371527      m_reihe   50000                   8    benzin             bmw   

       unrepaired_damage              date_ad  postal_code  \
0                    NaN  2016-03-24 00:00:00        70435   
1                     ja  2016-03-24 00:00:00        66954   
2                    NaN  2016-03-14 00:00:00        90480   
3                   nein  2016-03-17 00:00:00        91074   
4                   nein  2016-03-31 00:00:00        60437   
...                  ...                  ...          ...   
371523               NaN  2016-03-14 00:00:00        39576   
371524              nein  2016-03-05 00:00:00        26135   
371525              nein  2016-03-19 00:00:00        87439   
371526               NaN  2016-03-20 00:00:00        40764   
371527              nein  2016-03-07 00:00:00        73326   

                  last_seen  
0       2016-04-07 03:16:57  
1       2016-04-07 01:46:50  
2       2016-04-05 12:47:46  
3       2016-03-17 17:40:17  
4       2016-04-06 10:17:21  
...                     ...  
371523  2016-04-06 00:46:52  
371524  2016-03-11 18:17:12  
371525  2016-04-07 07:15:26  
371526  2016-03-24 12:45:21  
371527  2016-03-22 03:17:10  

[353943 rows x 17 columns]

## Exploring the Date columns

There are number of columns with date information:
- `date_crawled`: added by the crawler
- `last_seen`: added by the crawler
- `ad_created`: from the website
- `registration_month`: from the website
- `registration_year`: from the website


In [56]:
# How the values in three columns formatted
autos[["date_crawled", "date_ad", "last_seen"]][0:5]


date_crawled              date_ad            last_seen
0  2016-03-24 11:52:17  2016-03-24 00:00:00  2016-04-07 03:16:57
1  2016-03-24 10:58:45  2016-03-24 00:00:00  2016-04-07 01:46:50
2  2016-03-14 12:52:21  2016-03-14 00:00:00  2016-04-05 12:47:46
3  2016-03-17 16:54:04  2016-03-17 00:00:00  2016-03-17 17:40:17
4  2016-03-31 17:25:20  2016-03-31 00:00:00  2016-04-06 10:17:21

We can see that first 10 charecters represent the day, we can assume that information about the exact time when the ad was crawled is not so relevant, so to understand the date range we can just leave first 10 characters in each row using Series.str[:10], to include missing values in the distribution and to use percentages instead of counts we chain the `Series.value_counts(normalize=True, dropna=False) method`

In [57]:
# Calculate the distribution of values, including missing values
autos["date_crawled"].str[:10].value_counts(normalize=True, dropna=False).sort_index(ascending=False).head(10)

2016-04-07    0.001620
2016-04-06    0.003163
2016-04-05    0.012823
2016-04-04    0.037612
2016-04-03    0.038735
2016-04-02    0.035079
2016-04-01    0.034116
2016-03-31    0.031885
2016-03-30    0.033529
2016-03-29    0.034170
Name: date_crawled, dtype: float64

In [58]:
autos["date_crawled"].str[:10].value_counts(normalize=True, dropna=False).sort_values()

2016-04-07    0.001620
2016-04-06    0.003163
2016-04-05    0.012823
2016-03-18    0.013127
2016-03-06    0.014462
2016-03-13    0.015735
2016-03-05    0.025551
2016-03-24    0.029912
2016-03-16    0.030148
2016-03-27    0.030275
2016-03-17    0.031661
2016-03-31    0.031885
2016-03-26    0.031965
2016-03-23    0.031971
2016-03-22    0.032469
2016-03-10    0.032563
2016-03-11    0.032727
2016-03-25    0.032934
2016-03-15    0.033451
2016-03-08    0.033454
2016-03-30    0.033529
2016-04-01    0.034116
2016-03-29    0.034170
2016-03-09    0.034210
2016-04-02    0.035079
2016-03-28    0.035112
2016-03-19    0.035292
2016-03-07    0.035690
2016-03-21    0.035739
2016-03-12    0.036194
2016-03-14    0.036275
2016-03-20    0.036353
2016-04-04    0.037612
2016-04-03    0.038735
Name: date_crawled, dtype: float64

In [59]:
autos["date_ad"].str[:10].value_counts(normalize=True, dropna=False).sort_index(ascending=False).head(10)

2016-04-07    0.001558
2016-04-06    0.003155
2016-04-05    0.011655
2016-04-04    0.037741
2016-04-03    0.038893
2016-04-02    0.034966
2016-04-01    0.033976
2016-03-31    0.031955
2016-03-30    0.033381
2016-03-29    0.034221
Name: date_ad, dtype: float64

In [60]:
autos["last_seen"].str[:10].value_counts(normalize=True, dropna=False).sort_index(ascending=False).head(10)

2016-04-07    0.129648
2016-04-06    0.217830
2016-04-05    0.126206
2016-04-04    0.025654
2016-04-03    0.025366
2016-04-02    0.025016
2016-04-01    0.024022
2016-03-31    0.024238
2016-03-30    0.023856
2016-03-29    0.023312
Name: last_seen, dtype: float64

As we can see all the ads were relevant by the time it was crawled, crawlers were online last time between 2016-03-05 and 2016-04-07

In [23]:
autos["registration_year"].describe()

count    371528.000000
mean       2004.577997
std          92.866598
min        1000.000000
25%        1999.000000
50%        2003.000000
75%        2008.000000
max        9999.000000
Name: registration_year, dtype: float64

According to the information in `registration_year`
column, the oldest car were registered in 1000 which is not supposed to be true, because it is far before first car was invented, while maximum year 9999, that does not seem realistic too, as it is in future. So we have to correct the data in `registration_year` column.

## Dealing with Incorrect Registration data

Now it's time to clean wrong data in `registration_year` column. Obviously there shoudn't be any dates after the listing was seen and too early dates also are not suppose to be true. Data has been collected in 2016, so the latest date should be 2016 and we have to choose the earliest. That is not so easy, because technically there could be raritet cars in the list. let's choose 1916 as an earliest year, so the olders car would be 100 years old, that's more then enough.

Now, after we've left in our dataset only years between 1916 and 2016, let's calculate distribution of the remaining values and then sort it into descending order.

In [64]:
autos = autos[autos['registration_year'].between(1900, 2016)]
autos['registration_year'].value_counts(normalize=True, dropna=False).sort_index(ascending=True)


1910    0.000291
1911    0.000003
1915    0.000003
1919    0.000003
1920    0.000003
          ...   
2012    0.026397
2013    0.017257
2014    0.013459
2015    0.008358
2016    0.027633
Name: registration_year, Length: 97, dtype: float64

We can see that there are 97 unique registration years in the dataset: earliest date is in 1910 while the latest is 2016.

## Exploring Price by Brand

Now, we'll explore the unique values in `brand` and `price` columns

In [78]:
print(autos['brand'].value_counts(normalize=True))
print(autos['brand'].shape)

volkswagen        0.212391
bmw               0.109695
opel              0.107069
mercedes_benz     0.095955
audi              0.089352
ford              0.068849
renault           0.047559
peugeot           0.029859
fiat              0.025758
seat              0.018631
skoda             0.015407
mazda             0.015348
smart             0.014104
citroen           0.013874
nissan            0.013569
toyota            0.012747
sonstige_autos    0.010648
hyundai           0.009830
mini              0.009210
volvo             0.009129
mitsubishi        0.008266
honda             0.007587
kia               0.006878
alfa_romeo        0.006348
suzuki            0.006318
porsche           0.006149
chevrolet         0.005014
chrysler          0.003941
dacia             0.002450
jeep              0.002186
daihatsu          0.002186
land_rover        0.002136
subaru            0.002125
jaguar            0.001721
trabant           0.001620
saab              0.001452
daewoo            0.001438
r

`Volkswagen` is the most common brand with around 21.1 % of all the cars listed. This is almost twice the second placed brand `BMW` at 11 %. Opel, `Mercedes Benz` and `Audi` follow closely with 10.8 %, 9.6 %, and 8.7 % respectively. Top 5 brands are all German and represent over 60 % of the total listings. `Ford, Renault, Peugeot, Fiat, and Seat` make up the top 10.

There are a lot of brands that don't have a significant percentage of listings, so we will limit our analysis to only top 10 brands.



In [134]:
top_10_brands = autos['brand'].value_counts(normalize=True, dropna=False).head(10)
print(top_10_brands)
brand_mean_price = {}   #Empty dictionary to store mean price

for name in top_10_brands.index:
    brands = autos[autos['brand'] == name]
    mean_price = brands['price'].mean()
    brand_mean_price[name] = int(mean_price)
    
brand_mean_price

volkswagen       0.212391
bmw              0.109695
opel             0.107069
mercedes_benz    0.095955
audi             0.089352
ford             0.068849
renault          0.047559
peugeot          0.029859
fiat             0.025758
seat             0.018631
Name: brand, dtype: float64


{'volkswagen': 13643,
 'bmw': 14798,
 'opel': 3248,
 'mercedes_benz': 17614,
 'audi': 16218,
 'ford': 8702,
 'renault': 2366,
 'peugeot': 3206,
 'fiat': 4146,
 'seat': 4420}

On the top 10 brands, there is a distinct price gap:
 - Mercedes_benz, Audi anf BMW are more expensive
 - Ford ans Opel are less expensive 
 - Volkswagen is in between
 
 ## Exploring Correlation between Price and Mileage
 
 First, we'll create series object from `brand_price` dictionary by series constructor

In [120]:
brand_mean_mileage= {}
for brand in top_10_brands.index:
    brand_only = autos[autos['brand'] == brand]
    mean_mileage = brand_only['km'].mean()
    brand_mean_mileage[brand]= int(mean_mileage)
    
mean_mileage = pd.Series(brand_mean_mileage).sort_values(ascending=False)
mean_price = pd.Series(brand_mean_price).sort_values(ascending=False)
brand_info= pd.DataFrame(mean_mileage, columns=['mean_mileage'])
brand_info['mean_price'] = mean_price
brand_info

mean_mileage  mean_price
bmw                  132657       14798
mercedes_benz        130580       17614
audi                 129491       16218
opel                 128755        3248
volkswagen           128337       13643
renault              127875        2366
peugeot              124599        3206
ford                 123618        8702
seat                 120916        4420
fiat                 116519        4146

The range of car mileages does not vary as much as the prices do by brand, instead all falling within 10% for the top brands. There is a slight trend to the more expensive vehicles having higher mileage, with the less expensive vehicles having lower mileage.
    

In [125]:
brand_reg_year = {}

for brand in top_10_brands.index:
    brand_in = autos[autos['brand'] == brand]
    mean_year = brand_in['registration_year'].mean()
    brand_reg_year[brand] = int(mean_year)
brand_reg_year    

{'volkswagen': 2002,
 'bmw': 2002,
 'opel': 2002,
 'mercedes_benz': 2001,
 'audi': 2003,
 'ford': 2002,
 'renault': 2002,
 'peugeot': 2003,
 'fiat': 2002,
 'seat': 2004}

Most cars from the top 10 brands were registered between year 2002 - 2004. This shows that majority of the cars listed are at least 12 years old.

In [138]:
top_10_dist = {}

for b in top_10_brands.index:
    dist = top_10_brands[b]
    price_dist = round((dist * 100),1)
    top_10_dist[b] = price_dist

brand_dist = pd.Series(top_10_dist).sort_values(ascending=False)
brand_info['distribution(%)'] = brand_dist
brand_year = pd.Series(brand_reg_year).sort_values(ascending=False)
brand_info['registration_mean_year'] = brand_year
brand_info

mean_mileage  mean_price  distribution(%)  \
bmw                  132657       14798             11.0   
mercedes_benz        130580       17614              9.6   
audi                 129491       16218              8.9   
opel                 128755        3248             10.7   
volkswagen           128337       13643             21.2   
renault              127875        2366              4.8   
peugeot              124599        3206              3.0   
ford                 123618        8702              6.9   
seat                 120916        4420              1.9   
fiat                 116519        4146              2.6   

               registration_mean_year  
bmw                              2002  
mercedes_benz                    2001  
audi                             2003  
opel                             2002  
volkswagen                       2002  
renault                          2002  
peugeot                          2003  
ford                             2002  
seat                             2004  
fiat                             2002

Car listings on eBay Gemrany are dominated by the German brands. 
- Volkswagen is by far the most common brand with over 20 % dominance. 
- Most vehicles have a milages well over 125,000 km. 
- The prices are evenly distributed from 8700 to 14000.

Top 3 brands in price are over $15,000 per car. Majority of the vehcles were registered early 2000s.

## Conclusion

German e-bay. Audi, Mercedes Benz tand BMW are the most expencive and around 3 times more expencive then Ford and Opel, while Volkswagen is in between. We've checked if there is any correlation between the price difference and milage of top 6 brands, but we can observe that the mean mileage is almost the same for differtnt brands.
